pkg/asset/machines/aws/worker.go

```diff
@@ -2,98 +2,124 @@
 package aws
 
 import (
-	"text/template"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	awsprovider "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsproviderconfig/v1alpha1"
+	clusterapi "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 
 	"github.com/openshift/installer/pkg/types"
 )
 
-// WorkerConfig is used to generate the worker machinesets.
-type WorkerConfig struct {
-	Instances []WorkerMachinesetInstance
-	MachineConfig
-}
+// NonMasterMachineSets returns a list of machinesets for a machinepool.
+func NonMasterMachineSets(config *types.InstallConfig, pool *types.MachinePool, userDataSecret string) ([]clusterapi.MachineSet, error) {
```

Comment:
`NonMaster` -> [`Compute`][1] (or, if you prefer, `Worker`)?

[1]: https://github.com/openshift/installer/issues/264#issuecomment-422349518

Reply:
`NonMaster` is negative that means can use this for anything that is not master.

Reply:
> `NonMaster` is negative that means can use this for anything that is not master. ie don't care call it compute, worker, infra, whatever.

Do we care about the roles?  Can we make the functions `Machines` and `MachineSets` and add a `role` argument to use in the security-group, etc., labels?

Reply:
"Compute" would look good here IMO, I prefer that and "ControlPlane".